core/src/rpc.rs

```diff
@@ -2317,7 +2317,11 @@ impl RpcSol for RpcSolImpl {
         let (_, transaction) = deserialize_transaction(data, encoding)?;
 
         let mut result = if config.sig_verify {
-            transaction.verify()
+            if let Err(e) = transaction.verify() {
```

Comment:
Let's use the exact form as `send_transaction`:
```
            if transaction.verify().is_err() {
                return Err(RpcCustomError::TransactionSignatureVerificationFailure.into());
            }
            if let Err(e) = transaction.verify_precompiles() {
                return Err(RpcCustomError::TransactionPrecompileVerificationFailure(e).into());
            }
```
No need for possible return value differences between these two APIs.  Consider moving the verification into a shared function even

Reply:
Okay, good plan